src/NuGet.Core/NuGet.PackageManagement/Utility/FileSystemUtility.cs

```diff
@@ -8,14 +8,34 @@
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Xml.Linq;
 using NuGet.Common;
 using NuGet.PackageManagement;
 using NuGet.Packaging;
+using NuGet.Shared;
 
 namespace NuGet.ProjectManagement
 {
     public static class FileSystemUtility
     {
+        public static XDocument GetOrCreateXmlDocument(XName rootName, string root, string path, INuGetProjectContext nuGetProjectContext)
```

Comment:
Can this be made internal?  If not, I think I'd prefer that `NuGet.ProjectManagement.XmlUtility` is not deleted, but instead changed to be a wrapper of `NuGet.Shared.XmlUtility`, if that helps not needing this new public API.

Reply:
Good catch Andy. Changed the access modifier of `FileSystemUtility.GetOrCreateXmlDocument` method to `internal`.

Reply:
Consider `NuGet.PackageManagement.VisualStudio.dll` compiled against `NuGet.PackageManagement.dll` with your change, but then I overwrite `NuGet.PackageManagement.dll` with an older version. `NuGet.PackageManagement.VisualStudio.dll` will fail at runtime with a `MethodNotFoundException`. So, from a compatability point of view it is a new API.

I suggest keeping the old API signature (replace the body with this method's body), which would mean that changes to `src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Runtime/BindingRedirectManager.cs` can be undone.